### PR TITLE
Feat: add Prometheus and Grafana stack

### LINKED ISSUE #45: Deploy prometheus stack
Create two ArgoCD applications for monitoring:

* Prometheus
* Grafana

### LINKED ISSUE #49: Expose Grafana endpoint to the world
To access Grafana from the outside, it is proposed to create an endpoint called dash.cloudnativerioja.com

Be careful with the password and user by default

[Chart values](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack)

### LINKED ISSUE #52: Enable github SSO in Grafana
* Create OAUTH app in Github organization
* Populate the variables [here](https://github.com/grafana/helm-charts/blob/main/charts/grafana/values.yaml#L742) in the chart

### PR DESCRIPTION
**Pull request by statickidz**
# Prometheus and Grafana stack

Enhancing Monitoring and Visualization: This pull request introduces the Prometheus and Grafana stack, empowering our system with robust monitoring and dynamic visualization capabilities.

- Fixes #45
- Fixes #49 
- Fixes #52

## Public endpoint
- https://grafana.cloudnativerioja.com

## Github OAuth Spec

By default all user in the @cloudnativerioja organization are able to sign in but there are different roles for each team:
- @cloudnativerioja/organizers : [GrafanaAdmin](https://grafana.com/docs/grafana/latest/administration/roles-and-permissions/#grafana-server-administrators)
- @cloudnativerioja/collaborators : [Viewer](https://grafana.com/docs/grafana/latest/administration/roles-and-permissions/#grafana-server-administrators)

### CODE REVIEW

**Review by dcristobalhMad — APPROVED** [review]
This LGTM

File: applications/monitoring/grafana/grafana-operator.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            auth.github:
+              enabled: "true"
+              allow_sign_up: "true"
+              scopes: user:email,read:org
+              auth_url: https://github.com/login/oauth/authorize
+              token_url: https://github.com/login/oauth/access_token
+              api_url: https://api.github.com/user
+              allowed_organizations: cloudnativerioja
+              client_id: PLACEHOLDER
+              client_secret: PLACEHOLDER
```
[the remark this answers is not in the document]
Comment (edumgui) [reply applications/monitoring/grafana/grafana-operator.yaml]: There is already away in the README:

```yaml
grafana.ini:
  [auth.generic_oauth] enabled = true
  client_id = $__file{/etc/secrets/auth_generic_oauth/client_id}
  client_secret = $__file{/etc/secrets/auth_generic_oauth/client_secret}
```

[the remark this answers is not in the document]
Comment (statickidz) [reply applications/monitoring/grafana/grafana-operator.yaml]: @edumgui we manage to solve it by using secrets as environment vars with this keys `GF_AUTH_GITHUB_CLIENT_ID` `GF_AUTH_GITHUB_CLIENT_SECRET` and then referencing it from [here](https://github.com/cloudnativerioja/cluster-applications/pull/53/files#diff-99a6e7cd131de86bfbb160d7c5fdfcab106cbedf793d093e1e1d06f29d42fb2aR27):

```yaml
---
apiVersion: v1
kind: Secret
metadata:
  name: github-sso-secret
  namespace: monitoring
  labels:
    app.kubernetes.io/part-of: grafana
type: Opaque
stringData:
  GF_AUTH_GITHUB_CLIENT_ID: 123456789
  GF_AUTH_GITHUB_CLIENT_SECRET: 123456789
```

[the remark this answers is not in the document]
Comment (dcristobalhMad) [reply applications/monitoring/grafana/grafana-operator.yaml]: > There is already a way in the README:
> 
> ```yaml
> grafana.ini:
>   [auth.generic_oauth] enabled = true
>   client_id = $__file{/etc/secrets/auth_generic_oauth/client_id}
>   client_secret = $__file{/etc/secrets/auth_generic_oauth/client_secret}
> ```

Thanks @edumgui , you are right, I did not see it 🫠

**Pull request merged**